torch/ao/quantization/fx/_model_report/model_report.py

```diff
@@ -490,20 +522,63 @@ def generate_qconfig_mapping(self) -> QConfigMapping:
-                    is_activation_dynamic = current_options.is_activation_dynamic or detector_options.is_activation_dynamic
-                    is_per_channel = current_options.is_weight_per_channel or detector_options.is_weight_per_channel
-                    current_options.is_activation_dynamic = is_activation_dynamic
-                    current_options.is_weight_per_channel = is_per_channel
+                    update_qconfig_info_function(current_options, detector_options)
                 else:
                     # we just use this for now
                     detector_qconfig_info_compiled[module_fqn] = detector_info[module_fqn]
 
+        return detector_qconfig_info_compiled
+
+    def generate_qconfig_mapping(self) -> QConfigMapping:
+        r"""
+        Generates a QConfigMapping based on the suggestions of the
+        ModelReport API. The generated mapping encompasses all the
+        different types of feedback from the different detectors
+        all into one place.
+
+        These configs are based on the suggestions provided by the ModelReport API
+        and can only be generated once the reports have been generated.
+
+        Returns a QConfigMapping for the quantization configuration
+
+        Note:
+            Throws exception if we try to generate mapping on model we already removed observers from
+            Throws exception if we try to generate mapping without preparing for callibration
+        """
+        # get the mapping info
+        detector_qconfig_info_compiled = self._generate_qconfig_mapping(
+            self._update_detector_quantizaiton_qconfig_info
+        )
+
+        # we will do a bit of processing and remove fqns that don't have input weight recommended
+
         # now we generate the QConfig for each of the options
-        mapping: QConfigMapping = self._generate_qconfig_mapping_helper(detector_qconfig_info_compiled)
+        mapping: QConfigMapping = self._generate_qconfig_mapping_helper(
+            detector_qconfig_info_compiled,
+            self._quantization_config_generator
+        )
 
         # return the generated mapping
         return mapping
 
+    def _quantization_config_generator(self, detector_qconfig_info: DetectorQConfigInfo, module: torch.nn.Module) -> QConfig:
+        r"""
+        Returns the quantization configuration generated by the DetectorQConfigInfo object
+        """
+        return detector_qconfig_info.generate_quantization_qconfig(module)
+
+    def _equalization_config_generator(
+        self,
+        detector_qconfig_info: DetectorQConfigInfo,
+        module: torch.nn.Module
+    ) -> EqualizationQConfig:
+        r"""
+        We ignore the module argument here, and only focus on thedetector_qconfig_info
+
+        Returns the equalization configuration generated by the DetectorQConfigInfo object
+        """
+        return detector_qconfig_info.generate_equalization_qconfig()
+
     def generate_equalization_mapping(self) -> QConfigMapping:
```

Comment:
the name is equalization config I think: https://pytorch.org/docs/master/generated/torch.quantization.quantize_fx.prepare_fx.html#torch.quantization.quantize_fx.prepare_fx

Reply:
that's for the config file, but this is for the mapping that maps each module fqn to config